### PR TITLE
Extend drift detect server to expose metrics

### LINKED ISSUE #2537: Extend drift detector server (inside alibi detect server) to return metrics
Currently the CEServer in the alibi detect server has been extended to expose metrics returned. The custom metrics server is able to return metrics, which are then exposed. In order for the outlier detector server to also be able to expose metrics, the drift server will also have to be extended to enable it to return a `SeldonResponse` which could contain metrics server.

### PR DESCRIPTION
**Pull request by axsaucedo**
Fixes #2537

* Extends concept drift server to return metrics
* Locks alibi detect version to 0.4.1 due to breaking change on naming of UAE class https://github.com/SeldonIO/alibi-detect/blob/13bed2f70ba20e5c1360245f0c19a29dc1ae8889/alibi_detect/cd/preprocess.py#L14

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Oct 19 14:30:09 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=1

**Comment by seldondev** [comment]
Mon Oct 19 14:30:18 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=2

**Comment by seldondev** [comment]
Mon Oct 19 14:37:20 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=3

**Comment by seldondev** [comment]
Mon Oct 19 14:37:32 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=4

File: components/alibi-detect-server/adserver/cd_model.py
Comment (cliveseldon) [inline components/alibi-detect-server/adserver/cd_model.py:30]: Why is this a Gauge?

File: components/alibi-detect-server/adserver/cd_model.py
Comment (cliveseldon) [inline components/alibi-detect-server/adserver/cd_model.py:18]: Are these metrics keys not tied to a model or deployment? What happens if there are 2 in the same namespace. Any clashes?

File: components/alibi-detect-server/adserver/server.py
Comment (cliveseldon) [inline components/alibi-detect-server/adserver/server.py:239]: is ce_server not a bit generic. What is this argument used for?

**Review by cliveseldon — CHANGES_REQUESTED** [review]

File: components/alibi-detect-server/adserver/cd_model.py
Comment (axsaucedo) [reply components/alibi-detect-server/adserver/cd_model.py:18]: These are made unique by tags, which are added by the SeldonMetrics object - similar to the latency / request per second metrics

File: components/alibi-detect-server/adserver/cd_model.py
Comment (axsaucedo) [reply components/alibi-detect-server/adserver/cd_model.py:30]: The thought process is that feature_score is a metric related to a feature that changes on each request, as opposed to a total aggregate, would you say that COUNT would be more appropriate?

File: components/alibi-detect-server/adserver/server.py
Comment (axsaucedo) [reply components/alibi-detect-server/adserver/server.py:239]: This is an extra tag, more specifically it's the "method" tag, which in the case of the metrics in the python wrapper it would be "predict", "feedback", "input_transform", etc. In this case it would just be the "ce_server" metrics. I guess we could have it more specific to the type of server, such as DRIFT / OUTLIER / CUSTOM_METRICS?

File: components/alibi-detect-server/adserver/cd_model.py
Comment (axsaucedo) [reply components/alibi-detect-server/adserver/cd_model.py:18]: To be more specific, this uses the env variables that are set in the container (similar to the custom metrics server example), so you would have PREDICTOR_ID, etc.

**Comment by seldondev** [comment]
Wed Oct 21 13:50:07 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=5

**Comment by seldondev** [comment]
Wed Oct 21 13:50:24 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=6

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[cliveseldon](https://github.com/SeldonIO/seldon-core/pull/2557#issuecomment-714510593 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-714510593/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by cliveseldon — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[cliveseldon](https://github.com/SeldonIO/seldon-core/pull/2557#pullrequestreview-514752176 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu Oct 22 13:56:43 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=7

**Comment by seldondev** [comment]
Thu Oct 22 13:56:47 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2557/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2557 --build=8

**Pull request merged**
